### PR TITLE
DOC: Improve .at indexer API docs

### LINKED ISSUE #46722: DOC: KeyError isn't raised if df/series at property is called with label not existing in the df/series
### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/docs/dev/reference/api/pandas.DataFrame.at.html
https://pandas.pydata.org/docs/dev/reference/api/pandas.Series.at.html

### Documentation problem

Documentation states that `KeyError` would be raised if `at` was called on the DataFrame/Series with 'label' not existing in the data. It doesn't seem to be the case:
```
df = pd.DataFrame(data={'a':[0]})
df.at[0, 'b'] = 'no_error_raised'
df
Out[8]: 
   a                b
0  0  no_error_raised
```

```
srs = pd.Series([1, 2, 3], index=['a', 'b', 'c'])
srs.at['d'] = 'no_error_raised'
srs
Out[9]: 
a                  1
b                  2
c                  3
d    no_error_raised
dtype: object
```

Relates to #30649 - see the relevant comment:

> That line of docs was added in https://github.com/pandas-dev/pandas/pull/20290, and doesn't appear to be tested. It's not clear that that's the intentional behavior.

_Originally posted by @TomAugspurger in https://github.com/pandas-dev/pandas/issues/30649#issuecomment-570707381_

### Suggested fix for documentation

As stated by Tom, `it's not clear that that's the intentional behavior`. I can see two ways to fix it:

1. If it is decided that raising a KeyError is an intentional behavior, then some code changes would be required to make sure that it is indeed raised
2. If it is decided that KeyError should not be raised in such cases, documentation for both `dataframe.at` and `series.at` needs to be updated

IMO the wording in the docs `Access a single value for a row/column label pair.` suggests that it should indeed raise such an error as this should rather not be used to create new columns, but more discussion might be needed to reach conclusions

I am happy to tackle this issue if it is decided that it should be done:)

### PR DESCRIPTION
**Pull request by FactorizeD**
- [x] closes #46722 

Not sure about the wording of the `ValueError` and whether it is worth adding the additional methods under `See Also` but the same doc is used for both DataFrame and Series so I thought that mentioning all the methods is good for the sake of completeness

### CODE REVIEW

**Comment by jreback** [comment]
> under See Also but the same doc is used for both DataFrame and Series so I thought that mentioning all the methods is good for the sake of completeness

this sounds fine

File: pandas/core/indexing.py [hunk 1]
```diff
@@ -539,14 +539,25 @@ def at(self) -> _AtIndexer:
         Raises
         ------
         KeyError
-            If 'label' does not exist in DataFrame.
+            * If getting a value and 'label' does not exist in a DataFrame or
```
Comment (mroeschke) [inline pandas/core/indexing.py:542]: Mind Adding a `Notes` section linking to the user guide that was referenced in the original issue?

https://pandas.pydata.org/docs/dev/user_guide/indexing.html#fast-scalar-value-getting-and-setting

Comment (FactorizeD) [reply pandas/core/indexing.py:542]: Done

**Comment by FactorizeD** [comment]
Failing check seems not related tot his PR. Can this be merged in the current state after the check issue is resolved, or should I do something more?

File: pandas/core/indexing.py [hunk 2]
```diff
@@ -539,14 +539,31 @@ def at(self) -> _AtIndexer:
...
+            position(s).
+        Series.at : Access a single value by label.
+        Series.iat : Access a single value by integer position.
+        Series.loc : Access a group of rows by label(s).
+        Series.iloc : Access a group of rows by integer position(s).
+
+        Notes
+        -----
+        See `Fast scalar value getting and setting
+        <https://pandas.pydata.org/docs/dev/user_guide/indexing.html\
```
Comment (mroeschke) [inline pandas/core/indexing.py:565]: Could you use ``` :ref:`Fast scalar value getting and setting <indexing.basics.get_value>` ``` instead?

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @FactorizeD (sorry for the delay). Failure was unrelated

**Pull request merged**